services/web-app/components/page-nav/page-nav.js

```diff
@@ -0,0 +1,163 @@
+        contentRef.current?.querySelector(hash)?.scrollIntoView(true)
+      })
+    }
+  }, [handleHashChange, contentRef])
+
+  const checkIfSectionIdIsActive = (id) => {
+    const section = document.getElementById(id)
+    if (!section) return false
+    const sectionHeight = section.offsetHeight
+    const sectionBoundingClientRect = section.getBoundingClientRect()
+    const sectionTopDistance = sectionBoundingClientRect.top
+    // Space between top of screen and where we want section to be "active"
+    const scrollDistance = 90
+    const sectionIsAtTheTopOfTheView =
+      sectionTopDistance < scrollDistance && sectionHeight + sectionTopDistance - scrollDistance > 0
+
+    const sectionIsInView =
+      sectionBoundingClientRect.top >= 0 &&
+      sectionBoundingClientRect.left >= 0 &&
+      sectionBoundingClientRect.bottom <=
+        (window.innerHeight || document.documentElement.clientHeight) &&
+      sectionBoundingClientRect.right <= (window.innerWidth || document.documentElement.clientWidth)
+
+    const bottomReached =
+      document.documentElement.scrollTop + document.documentElement.clientHeight ===
+      document.documentElement.scrollHeight
+
+    const sectionActive = sectionIsAtTheTopOfTheView || (sectionIsInView && bottomReached)
+
+    if (sectionActive) {
+      setLastActiveLink(id)
+    } else {
+      setLastActiveLink(null)
+    }
+    return sectionActive
+  }
+
+  const setActiveLink = (link) => {
+    link.classList.add(styles.linkActive)
+
+    // Set url to current link as you scroll past
+    history.replaceState(null, null, `#${link.dataset.id}`)
+    setActiveItem(link.dataset.id)
+  }
+
+  const removeActiveLink = (link) => {
+    link.classList.remove(styles.linkActive)
+    if (activeItem === link.dataset.id) {
+      setActiveItem(null)
+      history.replaceState(null, null, window.location.pathname + window.location.search)
+    }
+  }
+
+  const setSelectedItem = () => {
+    const navLinks = Array.from(contentRef.current.querySelectorAll('[class^="page-nav_link"]'))
+    const sections = contentRef.current.querySelectorAll('[id]')
+
+    let activeSession = null
+    // Space between top of screen and where we want section to be "Active"
+    const scrollDistance = 90
+
+    sections.forEach((section) => {
+      const sectionHeight = section.offsetHeight
+      const sectionTopDistance = section.getBoundingClientRect().top
+      if (
+        // Setting active class when the top of the section is at the top
+        // of the screen and the bottom of the section is visible
+        sectionTopDistance < scrollDistance &&
+        sectionHeight + sectionTopDistance - scrollDistance > 0
+      ) {
+        activeSession = section
+      }
+    })
+
+    navLinks.forEach((link) => {
+      if (activeSession?.id === link.dataset.id) {
+        setActiveLink(link)
+      } else {
+        removeActiveLink(link)
+      }
+    })
+  }
+
+  const linkClicked = (id) => {
```

Comment:
minor: Some of the other event handlers are using `handle....()` as a naming convention. Could potentially use that here as well (i.e. `handleLinkClick`)

Reply:
Done!